### PR TITLE
Fix problems with url, pattern handling in mesheryctl onboard

### PR DESCRIPTION
**Pull request by nishantwrp**
Signed-off-by: Nishant Mittal <mittalnishant14@outlook.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: handlers/meshery_application_handler.go [hunk 1]
```diff
@@ -551,12 +551,17 @@ func githubRepoApplicationScan(
 	var mu sync.Mutex
 	ghWalker := walker.NewGit()
 	result := make([]models.MesheryApplication, 0)
+	fileName := path[strings.LastIndex(path, "/")+1:]
```
Comment (nishantwrp) [inline handlers/meshery_application_handler.go:554]: From the implementation it looks like this function was made considering that `path` will be a directory i.e. when a user did `mesheryctl app onboard -f https://github.com/service-mesh-patterns/service-mesh-patterns/tree/master/samples/applications/emojivoto` it onboarded all the files in that directory but failed when a user provided a specific file `mesheryctl app onboard -f https://github.com/service-mesh-patterns/service-mesh-patterns/blob/master/samples/applications/emojivoto/k8s.yaml`. 

I've changed the implementation assuming that the path provided will be a file only. Thoughts? // @Revolyssup @alphaX86 @gr455

**Comment by gr455** [comment]
@nishantwrp Alpha and myself were working on a PR that fixes this alongside a structural change with app import to avoid code duplication. I'll make that PR today, would you review that? You can fix the GitHub URLs part with another PR since that's not covered in the other PR.

I'm so sorry about the work duplication, I know that's disappointing

**Comment by nishantwrp** [comment]
That's fine @gr455, can you review the github urls part in this pr.

**Comment by gr455** [comment]
@nishantwrp Can you add your changes in https://github.com/meshery/meshery/pull/6061 (Also review this PR) 
@alphaX86 Can you give Nishant access?
Let's get all the fixes done on that PR

**Comment by alphaX86** [comment]
> @nishantwrp Can you add your changes in #6061 (Also review this PR) @alphaX86 Can you give Nishant access? Let's get all the fixes done on that PR

Providing access now...

**Comment by alphaX86** [comment]
@nishantwrp you can now work on PR #6061 and make changes there

**Comment by warunicorn19** [comment]
Hey @nishantwrp let's discuss this on the meshery dev call today.

**Comment by leecalcote** [comment]
merge conflicts...

**Comment by nishantwrp** [comment]
Closing in favour of #6061

**Pull request closed without merging**